src/main/java/org/vaadin/easyuploads/UploadField.java

```diff
@@ -573,41 +534,29 @@ public void removeListener(Property.ValueChangeListener listener) {
      * validated before the event is created.
      */
     protected void fireValueChange(boolean repaintIsNotNeeded) {
-        fireEvent(new AbstractField.ValueChangeEvent(this));
+        fireEvent(new HasValue.ValueChangeEvent(this, this, null, ! repaintIsNotNeeded));
```

Comment:
It may not be important, but could perhaps pass the old value, at least when it's known as in `setValue()` method. Not a blocker I guess.